pkg/deploy/tls/certificates.go

```diff
@@ -0,0 +1,174 @@
+	CheAllCACertsConfigMapName = "ca-certs-merged"
+	// CheMergedCAConfigMapRevisionsAnnotationKey is annotation name which holds versions of included config maps in format: cm-name1=ver1,cm-name2=ver2
+	CheMergedCAConfigMapRevisionsAnnotationKey = "che.eclipse.org/included-configmaps"
+
+	// Local constants
+	// labelEqualSign consyant is used as a replacement for '=' symbol in labels because '=' is not allowed there
+	labelEqualSign = "-"
+	// labelCommaSign consyant is used as a replacement for ',' symbol in labels because ',' is not allowed there
+	labelCommaSign = "."
+)
+
+type Certificates struct {
+}
+
+func NewCertificates() *Certificates {
+	return &Certificates{}
+}
+
+func (c *Certificates) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	if ctx.Proxy.TrustedCAMapName != "" {
+		if ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName == "" {
+			ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName = deploy.DefaultServerTrustStoreConfigMapName()
+			if err := deploy.UpdateCheCRSpec(ctx, "ServerTrustStoreConfigMapName", deploy.DefaultServerTrustStoreConfigMapName()); err != nil {
+				return reconcile.Result{}, false, err
+			}
+		}
+
+		result, done, err := c.syncTrustStoreConfigMapToCluster(ctx)
+		if !done {
+			return result, done, err
+		}
+	}
+
+	return c.syncAdditionalCACertsConfigMapToCluster(ctx)
+}
+
+func (c *Certificates) Finalize(ctx *deploy.DeployContext) error {
+	return nil
+}
+
+func (c *Certificates) syncTrustStoreConfigMapToCluster(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	name := ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName
+	configMapSpec := deploy.GetConfigMapSpec(ctx, name, map[string]string{}, deploy.DefaultCheFlavor(ctx.CheCluster))
+
+	// OpenShift will automatically injects all certs into the configmap
+	configMapSpec.ObjectMeta.Labels[injector] = "true"
+
+	actual := &corev1.ConfigMap{}
+	exists, err := deploy.GetNamespacedObject(ctx, name, actual)
+	if err != nil {
+		return reconcile.Result{}, false, err
+	}
+
+	if !exists {
+		// We have to create an empty config map with the specific labels
+		done, err := deploy.Create(ctx, configMapSpec)
+		return reconcile.Result{}, done, err
+	}
+
+	if actual.ObjectMeta.Labels[injector] != "true" {
+		actual.ObjectMeta.Labels[injector] = "true"
+		logrus.Infof("Updating existed object: %s, name: %s", configMapSpec.Kind, configMapSpec.Name)
+		if err := ctx.ClusterAPI.Client.Update(context.TODO(), actual); err != nil {
+			return reconcile.Result{}, false, err
+		}
+	}
+
+	return reconcile.Result{}, true, nil
+}
+
+func (c *Certificates) syncAdditionalCACertsConfigMapToCluster(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	// Get all source config maps, if any
+	caConfigMaps, err := GetCACertsConfigMaps(ctx.ClusterAPI.Client, ctx.CheCluster.GetNamespace())
+	if err != nil {
+		return reconcile.Result{}, false, err
+	}
+	if len(ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName) > 0 {
```

Comment:
```suggestion
        if ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName != "" {
```

Reply:
Don't we have a separate function `syncTrustStoreConfigMapToCluster` to handle it?
Maybe we should just add the required labels there and then the config map will be taken into account here automatically?